### PR TITLE
refactor: use OpenSea instead of footprint for floor prices

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

<del>Notice that other than removing footprint this pr only makes moralis the default provider for floor prices so we need to take in mind any custom env setting that overrides that value on any specific environment</del>

update: now footprint is replaced with opensea, it includes some of the changes from this pr https://github.com/ArdentHQ/dashbrd/pull/111/files to retrieve the opensea slug (cc @ItsANameToo)

Closes https://app.clickup.com/t/8678wj2dr and https://app.clickup.com/t/865d5et85

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: config/dashbrd.php [hunk 1]
```diff
@@ -67,7 +67,7 @@
         App\Jobs\FetchCollectionNfts::class => 'alchemy', // No other provider gives us `totalSupply` that we use in jobs...
         App\Jobs\FetchEnsDetails::class => 'moralis',
         App\Jobs\FetchNativeBalances::class => env('WEB3_PROVIDER_FETCH_NATIVE_BALANCES', 'moralis'),
-        App\Jobs\FetchCollectionFloorPrice::class => env('WEB3_PROVIDER_FETCH_NFT_COLLECTION_FLOOR_PRICE', 'mnemonic'),
+        App\Jobs\FetchCollectionFloorPrice::class => env('WEB3_PROVIDER_FETCH_NFT_COLLECTION_FLOOR_PRICE', 'moralis'),
```
Comment (alfonsobries) [inline config/dashbrd.php:70]: notice that im only changing the default value but still depends on `WEB3_PROVIDER_FETCH_NFT_COLLECTION_FLOOR_PRICE` env variable

**Review by ItsANameToo — DISMISSED** [review]

File: tests/App/Jobs/FetchCollectionFloorPriceTest.php [hunk 2]
```diff
@@ -8,9 +8,12 @@
 use App\Models\Token;
 use App\Support\Facades\Mnemonic;
 use Carbon\Carbon;
+use Illuminate\Support\Facades\Config;
 use Illuminate\Support\Facades\Http;
 
 it('should fetch nft collection floor price', function () {
+    Config::set('dashbrd.web3_providers.'.FetchCollectionFloorPrice::class, 'mnemonic');
```
Comment (oXtxNt9U) [inline tests/App/Jobs/FetchCollectionFloorPriceTest.php:15]: We should have a test for Moralis too especially if we now default to it. It doesn't look like we have one yet.

**Review by oXtxNt9U — CHANGES_REQUESTED** [review]

File: tests/App/Jobs/FetchCollectionFloorPriceTest.php (hunk 2 above)
Comment (alfonsobries) [reply tests/App/Jobs/FetchCollectionFloorPriceTest.php]: @oXtxNt9U no longer neccesary, updated the test to use opensea the new default

File: app/Http/Client/Opensea/OpenseaPendingRequest.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    /**
+     * @see https://docs.opensea.io/v1.0/reference/retrieving-collection-stats
+     */
+    public function getNftCollectionFloorPrice(string $collectionSlug): ?Web3NftCollectionFloorPrice
+    {
+        $response = self::get(sprintf('collection/%s/stats', $collectionSlug));
+
+        $floorPrice = $response->json('stats.floor_price');
+
+        $currency = 'eth';
```
Comment (ItsANameToo) [inline app/Http/Client/Opensea/OpenseaPendingRequest.php:80]: Is this also in eth for polygon collections?

File: .env.example [hunk 4]
```diff
@@ -96,6 +93,9 @@ COINGECKO_ENDPOINT=https://api.coingecko.com/api/v3/
 # Optional (only required if you want to use the PRO CoinGecko API)
 COINGECKO_API_KEY=
 
+# For a free API key just register an account at https://www.footprint.network/
```
Comment (ItsANameToo) [inline .env.example:96]: Needs adjusting as well

File: app/Http/Client/Opensea/OpenseaPendingRequest.php (hunk 3 above)
Comment (oXtxNt9U) [reply app/Http/Client/Opensea/OpenseaPendingRequest.php:80]: Still seems to be the case. 

Tested with the following collections which are in BNB and MATIC respectively but API reports them in ETH:

https://opensea.io/collection/meta-merge-pet
https://opensea.io/collection/yeah-skulls

**Review by oXtxNt9U — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**